src/config-file.c

```diff
@@ -259,12 +261,15 @@ Config* load_config_file(const gchar *config_file, GError **error)
                                                &config->auth_token, NULL, NULL);
         key_gateway_token_exists = get_key_string(ini_file, "client", "gateway_token",
                                                   &config->gateway_token, NULL, NULL);
-        if (!key_auth_token_exists && !key_gateway_token_exists) {
-                g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_INVALID_VALUE,
-                            "Neither 'auth_token' nor 'gateway_token' set");
+        key_client_cert_exists = get_key_string(ini_file, "client", "client_cert", &config->client_cert, NULL, NULL);
+
+        key_client_key_exists = get_key_string(ini_file, "client", "client_key", &config->client_key, NULL, NULL);
+
+        if (!key_auth_token_exists && !key_gateway_token_exists &&  !(key_client_cert_exists && key_client_key_exists)) {
+                g_set_error(error, 1, 4, "Neither a token nor client certificate are set!");
                 return NULL;
         }
-        if (key_auth_token_exists && key_gateway_token_exists) {
+        else if (key_auth_token_exists && key_gateway_token_exists) {
```

Comment:
As far as I can see, this allows various strange combinations of auth/gateway token and cert/key:
- auth_token + client_cert + client_key
- gateway_token + client_cert + client_key

Since there a no later checks, both auth methods would need to be present. Does hawkBit support these cases?

- auth_token + client_cert
- auth_token + client_key
- gateway_token + client_cert
- gateway_token + client_key

These cases lead to auth_token or gateway_token being used, but no warning/error. That should be fixed.

Reply:
Okay, this should be documented then. The last 4 scenarios should be fixed.

Reply:
Is this enough documented : https://github.com/rauc/rauc-hawkbit-updater/pull/166/files#diff-00d05b0405040981002c129a067f3ae336e345b5b62512322b116f63aec6c341R62 ? 
What do you mean by 'The last 4 scenarios should be fixed." ?